Add support for whpx accelerator to qemu builder

nelsonjchen opened the pull request:
Windows Hypervisor Platform (WHPX) is the Windows counterpart to HVF and
 KVM. It's an operating system provided component that provides
 virtualization acceleration support.

This is kind of the missing counterpart to https://github.com/hashicorp/packer/pull/6193.
QEMU 2.12 also added support for WHPX.

There's no support for libvirt on Windows so nothing was added in those
areas.

The popular QEMU for Windows distribution does not have WHPX support
built-in for legal reasons as the maintainer does not wish to use or
obtain any part of Microsoft's SDK to compile the distribution.
Workarounds and alternatives are documented.

vtolstov commented:
ci says that you need to go fmt you code ./builder/qemu/builder.go
if this is fixed - LGTM

nelsonjchen commented:
Fixed the `go fmt` issue. Updated the commit message to be a bit more clear as the reason WHPX isn't supported is because the QEMU for Window's maintainer doesn't want to use parts of the non-free or a not yet freed header part Windows 10 SDK to compile QEMU for Windows, not because it's not legal to use to cross-compile on Linux with copied WHPX headers.

nelsonjchen commented:
Oof! The orbital project just took off their building page. I'll need to adjust this.

https://github.com/AlexAltea/orbital/commit/f0fd82b1b7aa0fb18c089b11716731c3b628f504

SwampDragons commented:
Thanks for opening this PR!  I didn't even know qemu worked on windows at all.

SwampDragons commented on website/source/docs/builders/qemu.html.md.erb:
```diff
@@ -116,12 +116,19 @@ Linux server and have not enabled X11 forwarding (`ssh -X`).
+    -&gt; For `whpx`, note that [Stefan Weil's QEMU for Windows distribution](https://qemu.weilnetz.de/w64/)
+    does not include WHPX support and users may need to compile or source a
+    build of QEMU for Windows themselves with WHPX support. Users can refer
+    to these [instructions (Use on mainline QEMU repository instead)](https://github.com/AlexAltea/orbital/blob/f0fd82b1b7aa0fb18c089b11716731c3b628f504/docs/manual-windows.md)
```
I know you're trying to add those extra details to the docs (a link to instructions on how to build qemu with the necessary patch, and a binary with the patch included) in order to be helpful, but I think this kind of information is outside of the scope of Packer's docs. I'd be more comfortable sticking to just the first part, explaining that folks need to build QEMU themselves:

```-&gt; For `whpx`, note that [Stefan Weil's QEMU for Windows distribution](https://qemu.weilnetz.de/w64/)
    does not include WHPX support and users may need to compile or source a
    build of QEMU for Windows themselves with WHPX support.```

And trusting that users who want to use this feature are knowledgable enough about experimental QEMU features to roll their own.

nelsonjchen replied:
OK, that's reasonable. I'll make it so.
